src/ua_timer.c

```diff
@@ -139,7 +165,12 @@ UA_Timer_changeRepeatedCallbackInterval(UA_Timer *t, UA_UInt64 callbackId,
     /* Set the repeated callback */
     ZIP_REMOVE(UA_TimerZip, &t->root, te);
     te->interval = (UA_UInt64)(interval_ms * UA_DATETIME_MSEC); /* in 100ns resolution */
-    te->nextTime = UA_DateTime_nowMonotonic() + (UA_DateTime)te->interval;
+    UA_DateTime currentTime = UA_DateTime_nowMonotonic();
+    if(te->nextTime > (currentTime + UA_DATETIME_MSEC))
+        te->nextTime = te->nextTime + (UA_DateTime)te->interval;
+    else
+        te->nextTime = calculateNextTime_withPastBaseTime((currentTime + UA_DATETIME_MSEC), te->nextTime, (UA_DateTime)te->interval);
```

Comment:
I don't get the logic of this if/else.
You should use the timerPolicy to select how the nextTime is computed.

Put the timerPolicy and the basetime-pointer also in the API of UA_Timer_changeRepeatedCallbackInterval.
When changing the interval, we might change the basetime as well.